rfc/2020/alertmanager-integration/christyjacob4.md

```diff
@@ -0,0 +1,446 @@
+
+* Once this RFC is implemented, users will have the option to install **Alertmanager** in the control plane using a flag in in `linkerd install`
+
+
+- The optional **Alertmanager** installation will come with some default alerts
+  which users can configure to appropriate channels like slack, email, pagerduty, wechat etc.
+
+- Users will also be able to specify custom rules for Prometheus using a cli option.
+
+- Integration with Service Profiles will allow users to configure rules and alerts for per route metrics providing extremely fine grain alerting mechanisms for their apps.
+
+A POC for this has been done in [linkerd/linkerd2#1726](https://github.com/linkerd/linkerd2/pull/4124)
+
+# Design proposal (Step 2)
+
+The Integration of Alertmanager with the Existing Control Plane is a multi step process. It can be divided into the following stages
+
+---
+- ## Providing an option to include AlertManager in the installation
+The Linkerd install command is currently used to generate a helm chart based on various flags and options that are passed to it. The helm chart is then applied using `kubectl apply` .
+With the recent addition of [**Tracing**](https://github.com/linkerd/linkerd2/pull/3955) to linkerd installation as an add-on, we have a framework to add more modules as part of the Control Plane installation.
+
+The plan is to have an optional flag `--alertmanager` along with `linkerd install`, that would allow the user to install linkerd with Alertmanager configured out-of-the-box, as an addon. So the complete command would look like this 
+```sh
+linkerd install --alertmanager | kubectl apply -f -
+```
+This will simply enable Alertmanager with a default, publicly available [**webhook receiver**](#webhook-configuration) and configure Prometheus with a few default rules.  
+
+> This will enable the user to 
+>  * Check if the default Prometheus rules are firing.  
+>  * Check if Alertmanager is able to receive alerts from Prometheus.
+>  * Validate if Alertmanager is able to forward alerts to receivers.
+
+As an additional configuration item, a [`config.yaml`](#installation-yaml) can be passed to the `--alertmanager` flag. This would contain all the configurations required to setup a full feldged Alertmanager install. The documentation for the `config.yaml` file is given [**here**](#installation-yaml).
+
+For the Alertmanager to function properly, it should have the correct iptables and roles, this is done by the proxy injector and RBAC settings. More information about the implementation of the Alertmanager Helm chart can be found in this [**section**](#alertmanager-helm-chart). 
+
+---
+- ## Webhook configuration 
+
+[Webhook.site](https://webhook.site/) is an open source project that allows you to instantly get a unique, random URL that you can use to test and debug Webhooks and HTTP requests. Their API is public, free to use and doesn’t require authentication.  
```

Comment:
Hmm, This definitely provides a great out-of-box experience but I have some privacy concerns for users. I would be okay with having no receivers as the default! and maybe a flag to enable this?

Reply:
Ah, Nice! Looks like there is indeed a flag for it. Maybe we could have the default as false :)

Reply:
Yes i have addressed the privacy concerns in the proposal and provided a flag for it here https://github.com/linkerd/gsoc/pull/3/files#diff-c785814d0ceeb549aefcd289b04fec3eR114